### DISCUSSION TITLE
[SOLVED][nuxt3][quasar] Get Union Type of Routes Paths

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [posva/unplugin-vue-router#193] [Questions and Help] [answered]
Hi, I'm using Nuxt 3 with ` experimental: {typedPages: true}`.

I'm trying to find if there's an union type of all the routes paths,
so I can use it in a custom component extending Quasar components like QBtn which has the `to` attribute which accepts a route path.

So for example:

```ts
<script setup lang="ts">
import {QBtnProps} from 'quasar'
import { RoutePathUnion } from 'vue-router/auto'

interface AppBtn extends Omit<QBtnProps, "color"> {
    color?: "primary" | "secondary" | "negative"
    to?: RoutePathUnion
}
const props = defineProps<AppBtn>()
</script>

<template>
  <QBtn :color="props.color"></QBtn>
</template>

<style scoped>

</style>
```

And then having completion in `<AppBtn to="/some path"/>`.

Is there something like this, or is there a way I can implement this using `nuxt` package or `unplugin-vue-router` package APIs?

**Comment by angelhdzmultimedia** [comment]
### Solved

Haha thaaaanks Nuxt team!

```ts
type Routes = Parameters<typeof navigateTo>[0]


interface AppBtn extends Omit<QBtnProps, "color"> {
    color?: "primary" | "secondary" | "negative"
    to?: Routes
}
```

![image](https://github.com/posva/unplugin-vue-router/assets/10191085/d83108a1-db12-4de3-bfc8-a744532451eb)

**Comment by posva** [comment, marked as answer]
I never added the actual answer: these types are exported from `vue-router/auto`. See https://github.com/posva/unplugin-vue-router/issues/209#issuecomment-1661825831

**Reply by angelhdzmultimedia** [reply]
Yes thank you! I first found them in vue-router/auto, I forgot to mention this.

Then figured out I could just use navigateTo to extract the types from it. Good stuff!


Have a nice day.
